### ISSUE TITLE
Data format for training on THUMOS'14 dataset

### ISSUE THREAD

**Issue by dmvictor** [xumingze0308/TRN.pytorch#4]
According to the following structure, I need to prepare 3 .npy files for training on THUMOS'14 dataset. 

* THUMOS'14 dataset:
    ```
    $YOUR_PATH_TO_THUMOS_DATASET
    ├── resnet200-fc/
    |   ├── video_validation_0000051.npy (of size L x 2048)
    │   ├── ...
    ├── bn_inception/
    |   ├── video_validation_0000051.npy (of size L x 1024)
    |   ├── ...
    ├── target/
    |   ├── video_validation_0000051.npy (of size L x 22)
    |   ├── ...
    ```
In this [repo](https://github.com/yjxiong/anet2016-cuhk), we could do video classification with the two-stream model, which includes one reset-200 model for RGB input and one BN-Inception model for optical flow input. For reset-200 model, should I get the parameters from "caffe.Flatten_673" layer? I have checked the shape of the array, which is (1, 1, 2048). For BN-Inception model, should I get the parameters from "global_pool" layer? The shape of the array is (1, 10, 1024, 1, 1). Furthermore, I have no idea what .npy file I need to put in the "target" folder. Could you share what is the data format for training on THUMOS'14 dataset? Thank you!

**Comment by xumingze0308** [comment]
I think you should use the features after "global_pool" layer, whose size should be 1024. As it shows in the README, you should squeeze useless dimensions. The numpy file in the target folder stores the label of each frame in the video.

**Comment by dmvictor** [comment]
As you can see in [bn_inception_anet_2016_temporal_deploy.prototxt](https://github.com/yjxiong/anet2016-cuhk/blob/master/models/get_reference_models.sh), "global_pool" is the last layer before "fc-action". What do you mean "the features after global_pool layer"? As I mentioned in the question, the shape of the output is (1, 10, 1024, 1, 1). I have no idea which dimension is useless. How to manipulate the dimension "10" before "1024", could you elaborate a little more?
```
####################################### global pool #######################################
layer { name: "global_pool" top: "global_pool" bottom: "inception_5b/output" type: "Pooling"
  pooling_param { pool: AVE global_pooling: true stride: 1 } }
layer { name: "dropout" top: "global_pool" bottom: "global_pool" type: "Dropout"
  dropout_param { dropout_ratio: 0.7 } }

####################################### loss accuracy #######################################
layer { name: "fc-action" type: "InnerProduct" bottom: "global_pool" top: "fc-action"
  param { lr_mult: 1 decay_mult: 1 } param { lr_mult: 2 decay_mult: 0 }
  inner_product_param { num_output: 200
    weight_filler { type: "gaussian" std: 0.001 }
    bias_filler { type: "constant" value: 0 } } }
```

For the numpy file in the target folder, the size is L x 22. Is it storing the data like this: "Frame number" + ["20 action classes"](https://www.crcv.ucf.edu/THUMOS14/Class%20Index_Detection.txt) + "Ambiguous class"? Give [video_validation_0000051.mp4](https://www.crcv.ucf.edu/THUMOS14/Validation_set/videos/video_validation_0000051.mp4) as an example, the ground truth is "12 Billiards", which order is the third of the action classes. And the starting time of the action is second 2, and the ending time of the action is second 4. The data format of this video is something like the following: [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1], [1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1], [2 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0], [3 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0], [4 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0]...etc. Thanks for your response.

**Comment by xumingze0308** [comment]
(1) The shape of the output should have only four dimensions, where the first dimension (probably 10 in your case) is the batch size and the remaining (probably 1024 in your case) is the feature size.

(2) The 22 labels are "background class" + "20 action classes" + "ambiguous class", which is identical to image classification. You can find more details [here](https://github.com/xumingze0308/TRN.pytorch/blob/master/data/data_info.json).

**Comment by Dami111** [comment]
> (1) The shape of the output should have only four dimensions, where the first dimension (probably 10 in your case) is the batch size and the remaining (probably 1024 in your case) is the feature size.
> 
> (2) The 22 labels are "background class" + "20 action classes" + "ambiguous class", which is identical to image classification. You can find more details [here](https://github.com/xumingze0308/TRN.pytorch/blob/master/data/data_info.json).

Still have some doubts, could you help me? Targets for each video are chunk-level and every chunk has 6 frames. How could I use the 6 frames to get targets of the chunk?
